### ISSUE TITLE
Failed to locate the migration process output

### ISSUE THREAD

**Issue by dennisjalbertwork** [f5devcentral/f5-big-ip-migration-assistant#6]
After attempting a migration, it failed with the following error:
"Failed to locate the migration process output, cannot determine result of the migration- please check the device for errors"
I tried this on two different bigips with the same result.
Sources: 8900 & 3900 running 12.1.3.2
Destinations: vCMP guests running 12.1.3.2
I can't find any relevant logs on the destination device, but it doesn't say where to look either.

**Comment by JoeReifel** [comment]
hi @dennisjalbertwork, the log file it's looking for is located at /var/tmp/f5_migration_assistant.out

I can think of a couple normal reasons why this could occur: the first is that the management IP changed when the ucs file was loaded.  The second is that the admin password changed during the load.  Can both of these be ruled out?

**Comment by dennisjalbertwork** [comment]
Ok, that file doesn't exist on the destination device. 
Both of those reasons you mentioned are true. So are we not allowed to change the mgmt IP during the migration? I can try setting the admin pw to be the same and then try it again.

**Comment by JoeReifel** [comment]
The mgmt IP can change, all it means is the Migration Assistant can't connect to the device, since Migration Assistant is configured to query the IP and credentials you originally specified, not the new one in the ucs.  As long as the ucs load looks like it worked on the target BIG-IP, then the error from migration assistant can be considered normal and ignored.

**Comment by dennisjalbertwork** [comment]
Ok, I just tried it again after setting the same root & admin passwords on the source and destination devices. It failed with the same error. And there still isn't a log file to look at.

**Comment by JoeReifel** [comment]
Thanks for ruling out the simple item, I need some time to reproduce and I'll get back to you.  What version of the migration assistant are you running so I can compare apples to apples?  i'll assume 1.0.3, the latest.

**Comment by dennisjalbertwork** [comment]
Yes. 1.0.3

**Comment by JoeReifel** [comment]
Oops, I encountered #3 when connecting migration assistant to a VCMP instance.  That version won't work for me, I'll need to try with 1.0.4 beta.  Hopefully I can trigger the same symptom.

**Comment by dennisjalbertwork** [comment]
Any luck with this?

**Comment by JoeReifel** [comment]
Hi @dennisjalbertwork, sorry it's taken this long to respond, I didn't want to reply with "have you tried 1.04" when we hadn't tried it ourselves.  We're able to see the migration output on a platform migration on a 12.1.3.2 VCMP instance using F5 Migration Assistant v1.04, so now I'll ask - have you tried 1.04?

**Comment by dennisjalbertwork** [comment]
I just gave it a try. It was much faster this time. It still failed, but we got different results. This time the error said:
Configuration failed to load
Migration did not complete successfully- please refer to migration output for more details

When I clicked the button to view the migration details, it said the config failed to load so I logged into the device to look for additional details.

/var/tmp/f5_migration_assistant.out still doesn't exist, but I was able to find some useful info in the ltm log file:

notice mcpd[4814]: 01071029:5: Symmmetric Unit Key decrypt
notice mcpd[4814]: 01071027:5: Master key OpenSSL error: 1503872280:error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt:evp_enc.c:587: Z
notice mcpd[4814]: 01071038:5: Attempting Master Key migration to new unit key.
warning chmand[5316]: 012a0004:4: Store Read invalid store addr 0x3800, len 10
warning chmand[5316]: 012a0004:4: Store Read invalid store addr 0x3800, len 10
warning mcpd[4814]: 012a0004:4: halStorageRead: unable to read storage on this platform.
notice mcpd[4814]: 01071038:5: Unit key hash from key header: 94:3c:3b:f3:4f:57:05:cd:6e:ff:5b:6e:4c:83:8c:20:fc:bd:3d:7c
notice mcpd[4814]: 01071038:5: Unit key hash computed from read key:94:3c:3b:f3:4f:57:05:cd:6e:ff:5b:6e:4c:83:8c:20:fc:bd:3d:7c
notice mcpd[4814]: 01071038:5: Reloading the RSA unit to support config roll forward.
notice mcpd[4814]: 01071038:5: Loading keys from the file.
notice mcpd[4814]: 01390002:5: The size of the configuration DB has been extended by 5159040 bytes, now using a total of 26130560 bytes
notice mcpd[4814]: 01390002:5: The size of the configuration DB has been extended by 5159040 bytes, now using a total of 31289600 bytes
err mcpd[4814]: 0107102b:3: Master Key decrypt failure - decrypt failure - final
notice mcpd[4814]: 01071029:5: Master decrypt final
notice mcpd[4814]: 01071027:5: Master key OpenSSL error: 1503872280:error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt:evp_enc.c:587:
err mcpd[4814]: 01071769:3: Decryption of the field (cookie_encryption_passphrase) for object (/Common/cookie) failed.
err tmsh[20267]: 01420006:3: Loading configuration process failed.
emerg load_config_files: "/usr/bin/tmsh -n -g load sys config partitions all  platform-migrate" - failed. -- 01071769:3: Decryption of the field (cookie_encryption_passphrase) for object (/Common/cookie) failed. Unexpected Error: Loading configuration process failed.
warning mcpd[4814]: 01071423:4: High config load failed during a license reset. Fallback to base load.
info install_ucs.pm: Configuration loading error: high-config-load-failed
warning get_dossier[20373]: 01170017:4: Unable to read manifest (61).
info install_ucs.pm: ERROR: UCS installation failed.
err tmsh[18102]: 01420006:3: UCS loading process failed.

A tmsh load sys config shows similar results:
01071769:3: Decryption of the field (cookie_encryption_passphrase) for object (/Common/cookie) failed.
Unexpected Error: Validating configuration process failed.

I chose to keep the master encryption passphrase and copied it from the device which I migrated. It doesn't seem like it worked though.

**Comment by JoeReifel** [comment]
That's good news that it got further.  It sounds like root cause this time is an incorrect master key passphrase.  I can't help you with your passphrase but you could confirm the passphrase is correct by running the platform-migrate command manually on the BIG-IP in accordance with https://support.f5.com/csp/article/K82540512.

**Comment by dennisjalbertwork** [comment]
Ok, I've done a bunch more testing and I think I've proved the problem is with the migration utility.
I tried using the platform-migrate command like you suggested after the migration failure, but it has the same error about the cookie encryption passphrase. I then double checked my device names and IPs and ran the f5mku -K command on the source device again to make sure I had the right passphrase and everything was good. I wanted to make sure I could do a migration manually so I restored my new guest to a default config, ran re-key'ed it myself with f5mku -r xxxx and then loading the ucs manually. It gave me an error about the interfaces as expected due to the platform differences, but it seemed to get past the cookie encryption passphrase error which we've been seeing. I then reset the guest back to defaults, made sure to set the passphrase manually on the new device, and then used the tool thinking I could help it out with that part, but it seems like the tool overwrote they key which I set manually and messed it up again. It failed with the same cookie encryption passphrase error again. So the problem is that the tool is trying to rekey the device, but it isn't doing it properly with the key that I give it. It works fine if I don't use the tool and set the passphrase myself and load the config manually. If you want, we can create a new bug for this since it appears we have resolved the original issue and now have a new bug. Thanks for the help!
